bcache.c

```diff
@@ -49,26 +49,23 @@ char *C_MessageCachedir; ///< Config: (imap/pop) Directory for the message cache
  */
 struct BodyCache
 {
-  char path[PATH_MAX];
-  size_t pathlen;
+  char *path;
 };
 
 /**
  * bcache_path - Create the cache path for a given account/mailbox
  * @param account Account info
  * @param mailbox Mailbox name
- * @param dst     Buffer for the name
- * @param dstlen  Length of the buffer
+ * @param bcache  Body cache
  * @retval  0 Success
  * @retval -1 Failure
  */
-static int bcache_path(struct ConnAccount *account, const char *mailbox, char *dst, size_t dstlen)
+static int bcache_path(struct ConnAccount *account, const char *mailbox, struct BodyCache *bcache)
 {
   char host[256];
   struct Url url = { U_UNKNOWN };
-  int len;
 
-  if (!account || !C_MessageCachedir || !dst || (dstlen == 0))
+  if (!account || !C_MessageCachedir || bcache)
```

Comment:
The conditional is going the wrong way. Should be `!bcache` https://github.com/neomutt/upstream-mutt/commit/ad72ec6c#diff-dd9883995d9854abd43600c8140f2d3bR46

Reply:
Argh!  Thanks.